impl/asm-based-transformer/src/main/java/org/wildfly/transformer/asm/TransformerImpl.java

```diff
@@ -226,6 +228,86 @@ public void visitMethodInsn(int opcode, String owner, String name, String desc,
+                        final String bootstrapMethodHandleDescOrig = bootstrapMethodHandle.getDesc(); 
+                        String bootstrapMethodHandleDesc = replaceJavaXwithJakarta(bootstrapMethodHandleDescOrig);
+                        if (!descOrig.equals(desc)) {  // if we are changing
+                            // mark the class as transformed
+                            setClassTransformed(true);
+                        }
+                        if (!ownerOrig.equals(bootstrapMethodHandleOwner) ||
+                        !bootstrapMethodHandleDescOrig.equals(bootstrapMethodHandleDesc)) {  // if we are changing
+                            // mark the class as transformed
+                            setClassTransformed(true);
+                            bootstrapMethodHandle = new Handle(
+                                    bootstrapMethodHandle.getTag(),
+                                    bootstrapMethodHandleOwner,
+                                    bootstrapMethodHandle.getName(),                                     
+                                    bootstrapMethodHandleDesc,
+                                    bootstrapMethodHandle.isInterface());
+                        }
+                        /** bootstrapMethodArguments can be Integer, Float,Long, Double,String, Type, Handle, ConstantDynamic value. 
+                         * ConstantDynamic can modify content of array at runtime.
+                         */
+                        Object[] copyBootstrapMethodArguments = null;
+                        for(int looper = 0; looper < bootstrapMethodArguments.length; looper++) {
+                            Object argument = bootstrapMethodArguments[looper];
+                            if (argument instanceof Type) {
+                                Type type = (Type) argument;
+                                String oldDesc = type.getDescriptor();
+                                String updatedDesc = replaceJavaXwithJakarta(type.getDescriptor());
+                                if (!oldDesc.equals(updatedDesc)) {
+                                    setClassTransformed(true);
+                                    if (copyBootstrapMethodArguments == null) {
+                                        copyBootstrapMethodArguments = cloneBootstrapMethodArguments(bootstrapMethodArguments);
+                                    }
+                                    copyBootstrapMethodArguments[looper] = Type.getMethodType(updatedDesc);
+                                }
+                            } else if (argument instanceof Handle) {  // reference to a field or method
+                                Handle handle = (Handle)argument;
+                                String origDesc = handle.getDesc();
+                                String updatedDesc = replaceJavaXwithJakarta(handle.getDesc());
+                                if (!origDesc.equals(updatedDesc)) {  // if we are changing
```

Comment:
Also handle.getOwner() should be inspected and potentially replaced if mapping applies to it.